core/dom/element.js

```diff
@@ -2243,15 +2243,22 @@ CKEDITOR.dom.element.clearMarkers = function( database, element, removeFromDatab
 	 * Sets the element size considering the box model.
 	 *
 	 * @param {'width'/'height'} type The dimension to set.
-	 * @param {Number} size The length unit in px.
+	 * @param {Number/String} size The length unit.
 	 * @param {Boolean} isBorderBox Apply the size based on the border box model.
 	 */
 	CKEDITOR.dom.element.prototype.setSize = function( type, size, isBorderBox ) {
 		if ( typeof size == 'number' ) {
 			if ( isBorderBox && !( CKEDITOR.env.ie && CKEDITOR.env.quirks ) )
 				size -= marginAndPaddingSize.call( this, type );
 
-			this.setStyle( type, size + 'px' );
+			this.setStyle( type, CKEDITOR.tools.cssLength( size ) );
+		} else if ( typeof size == 'string' ) {
+			size = CKEDITOR.tools.convertToPx( size );
+
+			if ( typeof size != 'string' && isBorderBox && !( CKEDITOR.env.ie && CKEDITOR.env.quirks ) )
```

Comment:
Is `typeof size != 'string' && ` condition required at this point?

Reply:
Yes, if user passes percent value as width then `size` is still type of string. I can simplify it.